bridgebot/bidding.py

```diff
@@ -1,21 +1,123 @@
-from enums import Players, Strains, AuctionStatus, Doubles, Contracts, PASS, Team
-
+from enums import Players, Strains, AuctionStatus, Doubles, Contracts, Team, InvalidPlayerException, InvalidStrainException
+from enum import Enum
 
 class InvalidDealerException(Exception):
     pass
 
+class InvalidBidException(Exception):
+    pass
 
 class Contract:
     def __init__(self):
-        self.level = 0
-        self.strain = Strains.NT
+        self.contract = None
         self.doubled = Doubles.NONE
         self.declarer = None
+        self.last_bid_by = None # Redundant, but cleans up edge case logic in Bids.is_sufficient_bid
+
+
+# Compares to a contract, so can't go into enum
```

Comment:
Not sure whether to put this into `enums.py` or not. It needs the `Contract` class for some of its member functions, so (to me at least) this seems like an intelligent place for it.